### DISCUSSION TITLE
Is there a way to define schema locations globaly?

### DISCUSSION THREAD

**Discussion by LokiMidgard** [redhat-developer/vscode-xml#870] [Q&A] [answered]
I have diffrent files in different folders that all use the same set of local xsd files.

Is there a way to define a project wide setting for the location?

Copy pasting the schama location in every file is tideos aspecial since I need to change the amount of `../`

**Comment by fbricon** [comment]
Sounds like File association would fix your problem: https://github.com/redhat-developer/vscode-xml/blob/main/docs/Validation.md#xml-file-association-with-xsd

**Reply by LokiMidgard** [reply]
When I saw fileAssociations I thoght It allowed to add different file extensions and did not read the doc to that…

Thank you I would have nover foud it i guess 😳

**Reply by fbricon** [reply]
Yeah maybe "schema association " would be better. WDYT @angelozerr @rgrunber

**Reply by LokiMidgard** [reply]
I think it was actully not what I needed, but the doc you linked had the answer.

If I understood correctly file assositaion would associate a file with one xsd. I use several xsd files. But [XML catalog with XSD](https://github.com/redhat-developer/vscode-xml/blob/main/docs/Validation.md#xml-catalog-with-xsd) seems to be what I need :)

**Reply by angelozerr** [reply]
> Yeah maybe "schema association " would be better. WDYT @angelozerr @rgrunber

For DTD, it is not a schema. I think better name should be grammarAssociation. But today there are more and more user who uses this settings, it can b eannoying to rename it.

**Reply by fbricon** [reply]
@angelozerr we can easily automate the migration if needed

**Reply by LokiMidgard** [reply]
for information I looked at every entry under `xml.validation.*` but that was probably missguided.

**Comment by LokiMidgard** [comment, marked as answer]
Thanks to @fbricon I found the docu to [XML catalog with XSD](https://github.com/redhat-developer/vscode-xml/blob/main/docs/Validation.md#xml-catalog-with-xsd).
